src/js/util.js

```diff
@@ -62,7 +67,7 @@ function findSingleNodeValue(xpath, elem) {
             null
         ).singleNodeValue;
     } catch (ex) {
-        console.warn('findSingleNodeValue blew up with: ' + xpath);
+        console.warn('findSingleNodeValue blew up with: ', xpath);
```

Comment:
we need to reduce the number of warnings in the log - and this line (both your version and mine) is a strong candidate. I reckon it should only warn when all of the choices have failed to work.

Reply:
(no need to fix in this review though - just thinking out loud)